src/transfer/stores/headless.ts

```diff
@@ -80,35 +96,117 @@ export default class HeadlessStore implements IHeadlessStore {
 
   @action
   transferGold = async (
+    signer: string,
     recipient: string,
     amount: Decimal,
     memo: string
   ): Promise<TxId> => {
-    this.assertAgentAddress();
+    if (signer.startsWith("0x")) {
+      signer = signer.substr(2);
+    }
+    if (recipient.startsWith("0x")) {
+      recipient = recipient.substr(2);
+    }
 
-    const nextTxNonceData = await this.graphqlSdk.GetNextTxNonce({
-      address: this.agentAddress,
-    });
-    if (!nextTxNonceData.data) {
-      throw new Error("Failed to get next nonce");
+    if (signer === recipient) {
+      const errorMessage = "You can't transfer NCG to yourself.";
+      alert(errorMessage);
```

Comment:
I think we avoid UI interaction on the store level. how about moving this `alert()` to its caller?